File: src/app/java/CorruptionException.java
package app.java;

public class CorruptionException extends Throwable {
  public CorruptionException(String message) {
    super(message);
    message.getClass();
  }

  public CorruptionException(String message, Throwable cause) {
    super(message, cause);
    cause.getClass();
  }

  @Override
  public void printStackTrace() {
    System.out.println("CorruptionException: " + new CorruptionException("").getMessage());
  }
}

File: kt/app/java/CorruptionException.kt
package app.java

import app.java.CorruptionException

class CorruptionException : Throwable {
    constructor(message: String) : super(message) {
        message.javaClass
    }

    constructor(message: String?, cause: Throwable) : super(message, cause) {
        cause.javaClass
    }

    override fun printStackTrace() {
        println("CorruptionException: " + CorruptionException("").message)
    }
}